torch/cuda/comm.py

```diff
@@ -90,6 +92,8 @@ def reduce_add(inputs, destination=None):
     assert nccl_root is not None, "reduce_add expects destination to be on the same GPU with one of the tensors"
     with torch.cuda.device(destination):
         result = type(inp)(input_size).zero_()
+        if is_sparse:
+            result.resize_as_(inp) # this is needed to get correct dimensonV
```

Comment:
dimensionV

Reply:
Maybe use `type(inp).resize_as_(inp)` in all cases?